server.go

```diff
@@ -560,18 +560,24 @@ func (srv *Server) serveDNS(m []byte, w *response) {
 	req := new(Msg)
 	req.setHdr(dh)
 
-	switch srv.MsgAcceptFunc(dh) {
+	switch action := srv.MsgAcceptFunc(dh); action {
 	case MsgAccept:
 		if req.unpack(dh, m, off) == nil {
 			break
 		}
 
 		fallthrough
-	case MsgReject:
+	case MsgReject, MsgRejectNotImplemented:
+		opcode := req.Opcode
 		req.SetRcodeFormatError(req)
+		req.Zero = false
+		if action == MsgRejectNotImplemented {
```

Comment:
I removed the `SetRcodeNotImplemented` function. In the end, in light of the `opcodeflg` test, a NotImplemented message is basically a message with QR set, opcode set to original opcode, rcode set to NotImpl and all other flags cleared.
I think it could be valuable to have `SetRcodeNotImplemented` dealing with this.